spec/requests/admin/users_spec.rb

```diff
@@ -17,67 +17,134 @@
     end
 
     context 'ログインしている場合' do
+      context 'ユーザが管理者ではない場合' do
+        subject { get admin_users_path }
+
+        let(:non_admin_user) { create(:user, :noadmin) }
+
+        before { log_in_as(non_admin_user) }
+
+        it 'ログインページにリダイレクトしてトーストメッセージを表示する' do
+          expect(subject).to redirect_to login_url
+          expect(response).to have_http_status :see_other
+          expect(flash[:danger]).to be_present
+        end
+      end
+
       context 'ユーザが管理者の場合' do
+        let(:order_by) { 'asc' }
+        let(:sort_key) { 'name' }
+        let(:limit) { 10 }
+        let(:access_token) { AccessToken.new(email: admin_user.email).encode }
+        let(:admin_user) { create(:user, :admin) }
+
+        before do
+          allow(Api::AccessToken).to receive(:create).and_return(Api::AccessToken.new(value: access_token))
+          post login_path,
+               params: { session: { email: admin_user.email, password: admin_user.password, remember_me: '1' } }
+        end
+
         context '例外が返ってきた場合' do
-          xit 'ホーム画面にリダイレクトして、取得に失敗した旨をトーストメッセージで表示する' do
-            # TODO: specの内容を作成する
+          subject { get admin_users_path }
+
+          before { allow(Api::User).to receive(:get_list).and_raise(StandardError) }
+
+          it 'ホーム画面にリダイレクトして、取得に失敗した旨をトーストメッセージで表示する' do
+            expect(subject).to redirect_to root_url
+            expect(Api::User).to have_received(:get_list)
+            expect(response).to have_http_status :see_other
+            expect(flash[:danger]).to be_present
           end
         end
 
         context 'ユーザの配列が返ってきた場合' do
+          before do
+            user_list = create_list(:user, 20)
+            api_users = user_list.take(limit).map do |user|
+              Api::User.new(
+                id:           user.id,
+                name:         user.name,
+                email:        user.email,
+                admin:        user.admin,
+                activated:    user.activated,
+                activated_at: user.activated_at&.iso8601(2),
+                created_at:   user.created_at.iso8601(2),
+                updated_at:   user.updated_at.iso8601(2)
+              )
+            end
+
+            allow(Api::User).to receive(:get_list).and_return(api_users)
+          end
+
           context 'クエリパラメータにpageがない場合' do
-            xit 'ユーザー取得の関数にoffset=0パラメータをつけて呼び出す' do
-              # TODO: specの内容を作成する
+            subject { get admin_users_path }
+
+            it 'ユーザー取得の関数にoffset=0パラメータをつけて呼び出す' do
+              subject
+              expect(Api::User).to have_received(:get_list).with(access_token:, limit:, offset: 0)
             end
 
-            xit 'ステータスコード200とともに、10件分のユーザーを表示する' do
-              # TODO: specの内容を作成する
+            it 'ステータスコード200とともに、10件分のユーザーを表示する' do
+              subject
+              expect(response).to be_successful
```

Comment:
僕たちがよく使うテクニックで、subject の返り値を response にしてしまうというものがあります。

```
subject {
  get admin_users_path
  response
}
```

こうすることで、subject を参照するだけでよくなるため、spec がちょっぴりシンプルになります。
```suggestion
              expect(subject).to be_successful
```